cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs

```diff
@@ -1,21 +1,30 @@
+#[cfg(feature = "SGX_MODE_HW")]
+use enclave_ffi_types::NodeAuthResult;
 ///
 /// These functions run off chain, and so are not limited by deterministic limitations. Feel free
 /// to go crazy with random generation entropy, time requirements, or whatever else
 ///
 use log::*;
+#[cfg(feature = "SGX_MODE_HW")]
+use sgx_types::{sgx_platform_info_t, sgx_update_info_bit_t};
 use sgx_types::{sgx_quote_sign_type_t, sgx_status_t};
-
 use std::slice;
 
 use crate::consts::{
     ATTESTATION_CERTIFICATE_SAVE_PATH, ENCRYPTED_SEED_SIZE, IO_CERTIFICATE_SAVE_PATH,
     SEED_EXCH_CERTIFICATE_SAVE_PATH,
 };
 use crate::crypto::{Keychain, KEY_MANAGER, PUBLIC_KEY_SIZE};
+#[cfg(feature = "SGX_MODE_HW")]
+use crate::registration::report::AttestationReport;
 use crate::storage::write_to_untrusted;
 use crate::utils::{attest_from_key, validate_const_ptr, validate_mut_ptr, validate_mut_slice};
 
 use super::attestation::create_attestation_certificate;
+#[cfg(feature = "SGX_MODE_HW")]
+use super::cert::ocall_get_update_info;
+#[cfg(feature = "SGX_MODE_HW")]
+use super::cert::verify_quote_status;
```

Comment:
You can probably compress these imports that require `#[cfg(feature = "SGX_MODE_HW")]` under fewer clauses.
at least everything from this crate can be combined so you use fewer `cfg`s